src/common/hooks/use-recently-viewed.tsx

```diff
@@ -0,0 +1,21 @@
+import { useState, useEffect } from 'react';
+import { store } from '@common/utils';
+
+const storageKey = 'recentlyViewedTx';
+
+export const useRecentlyViewed = (txid?: string) => {
+  if (!process.browser) return;
+
+  const storedTxs = store.get(storageKey, []);
+  const [viewedTransactions, setViewedTransactions] = useState(storedTxs);
+
+  useEffect(() => {
+    if (txid) {
+      const newTxList = [...viewedTransactions, txid];
+      setViewedTransactions(newTxList);
+      store.set(storageKey, newTxList);
+    }
+  }, [txid]);
```

Comment:
We aren't doing anything to filter duplicates. Might be better to check if exists, and push to the end of the array instead of adding a dupe.

Reply:
I'd also maybe rename it to `use-recently-viewed-tx`/`useRecentlyViewedTx`

Reply:
Yep, good spot. This isn't quite final yet as will need to save more than just txid, per designs.